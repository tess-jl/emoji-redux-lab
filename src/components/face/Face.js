import React from 'react';
import styles from './Face.css';
import { useSelector } from 'react-redux';
import { getFace } from '../../selectors/moodSelectors';

const Face = () => {
  const emojiFace = useSelector(getFace);

  return (
    <p className={styles.Face}>
      {emojiFace}
    </p>
  );
};

export default Face;
