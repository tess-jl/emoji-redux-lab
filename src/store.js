const { createStore } = require('redux'); 
import reducer from './reducers/moodsReducer';

export default createStore(
  reducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);
