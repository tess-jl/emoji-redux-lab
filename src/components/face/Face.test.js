import React from 'react';
import { shallow } from 'enzyme';
import Face from './Face';
import { Provider } from 'react-redux';
import store from '../../store';

describe('Face component', () => {
  it('renders Face', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Face />
      </Provider>
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
  
