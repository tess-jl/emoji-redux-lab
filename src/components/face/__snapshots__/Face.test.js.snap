// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Face component renders Face 1`] = `<Face />`;
